RestaurantsNearMe/Data/APIKeyProvider.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  APIKey.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-09.
+//
+
+import Foundation
+import Combine
+import CloudKit
+
+struct APIKeyProvider {
+  static var key: APIKey?
```

Comment:
Once the key is fetched, it'll be accessed via a static object (kind of like a singleton)

Reply:
I understand you'd like to build this out as your own, but consider the `swift-dependencies` library. It's built for exactly these types of things!

Alternatively, you could write your own "light" version with a custom property wrapper.